lib/kube/proxy/forwarder.go

```diff
@@ -858,7 +872,27 @@ func (f *Forwarder) setupContext(ctx context.Context, authCtx auth.Context, req
 				),
 			)
 			defer span.End()
-			return new(net.Dialer).DialContext(ctx, network, endpoint.addr)
+			conn, err := new(net.Dialer).DialContext(ctx, network, endpoint.addr)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			// Make sure we will only send signed PROXY headers to teleport kube service node, not real kube cluster
+			if forwarderType != ProxyService && endpoint.serverID == "" {
+				clientDst = nil
+			}
+			// Teleport proxy will send signed PROXY headers to the kube service
```

Comment:
Can we simplify a bit this block of code, if we wrapped the `net.Dialer` into a dalier that does the dial and writes the proxy signature if `proxySigner!=nil` and `clientSrc!=nil`?

I am not sure if other protocols would use something similar but at least will simplify this code.
what do you think @rosstimothy @AntonAM ?

Reply:
I can look into this as a separate change 👌